controllers/controllers/services/instances/upsi/controller.go

```diff
@@ -133,11 +133,29 @@ func (r *Reconciler) ReconcileResource(ctx context.Context, cfServiceInstance *k
 	}
 
 	if err = r.validateCredentials(credentialsSecret); err != nil {
+		cfServiceInstance.Status.LastOperation = korifiv1alpha1.LastOperation{
+			Type:  "create",
+			State: "failed",
+		}
 		return ctrl.Result{}, k8s.NewNotReadyError().WithCause(err).WithReason("SecretInvalid")
```

Comment:
We should not requeue this error as an invalid secret can only be fiexed by the client. We should return `NewNotReadyError. ... .WithNoRequeue`. As this is hard to test and the test's value is arguable we can skip the test in this case.